scripts/urban.js

```diff
@@ -6,68 +6,49 @@
 //
 
 module.exports = function (robot) {
-	robot.respond(/!?urban (.+)/i, function (res) {
-		robot.getUrban(res, res.match[1].trim());
-	});
-
-	robot.getUrban = function (res, word) {
-
-		robot.http("http://api.urbandictionary.com/v0/define?term=" + word).get()
-		(function (err, resp, body) {
-			if (!err) {
-				var response = "";
-
-				var max = 2;
-				var lines = 0;
-
-				if (robot.getUrbanResult(body) == "no_results") {
-					response += "> :( No result found for " + word;
-				} else {
-					var definition = robot.getUrbanDef(body);
-					var example = robot.getUrbanExample(body);
-					response += ">" + word.toUpperCase() + ": ";
-					definition.split("\r\n").forEach(function (line) {
-						if (lines <= max) {
-							if (lines == 0) {
-								response += line + "\r\n";
-							} else {
-								response += "> " + line + "\r\n";
-							}
-						}
-						lines++;
-					});
-					example.split("\r\n").forEach(function (line) {
-						if (lines <= max) {
-							response += "> \t _" + line + "_ \r\n";
-						}
-						lines++;
-					});
-				}
-
-				if (lines > 2) {
-					response += " - more at http://www.urbandictionary.com/define.php?term=" + encodeURI(word);
-				}
-
-				res.send(response);
-			}
-		});
-	};
-
-	// Parses the JSON to get the first definition
-	robot.getUrbanResult = function (string) {
-		json = JSON.parse(string);
-		return json["result_type"];
-	};
-
-	// Parses the JSON to get the first definition
-	robot.getUrbanDef = function (string) {
-		json = JSON.parse(string);
-		return json["list"][0]["definition"];
-	};
-
-	// Parses the JSON to get the first definition
-	robot.getUrbanExample = function (string) {
-		json = JSON.parse(string);
-		return json["list"][0]["example"];
-	};
+  robot.respond(/!?urban(.*)/i, function (res) {
+    var phrase = res.match[1].trim();
+    var urban = robot.http("http://api.urbandictionary.com/v0/define?term=" + encodeURI(phrase))
+      .get()(function (err, resp, body) {
+
+        // Check for correct usage
+        if (res.match[1][0] !== " " || !phrase || phrase.length < 1) {
+          return res.send("> Usage: `!urban <SEARCH_PHRASE>`");
+        }
+
+        // Check for JS/hubot errors
+        if (err) {
+          return res.send(">>> Error: " + err.toString() + ".");
+        }
+
+        // Check for HTTP Errors
+        if (resp.statusCode !== 200) {
+          return res.send("> HTTP Error ( " + resp.statusCode + ").");
+        }
+
+        var udResp = JSON.parse(body);
```

Comment:
maybe give a description for this variable or choose a better name as its a bit unclear